core/blockchain_test.go

```diff
@@ -3188,6 +3188,9 @@ func TestTransactionCountLimit(t *testing.T) {
 	config := params.TestChainConfig
 	config.Scroll.MaxTxPerBlock = new(int)
 	*config.Scroll.MaxTxPerBlock = 1
+	defer func() {
+		config.Scroll.MaxTxPerBlock = nil
+	}()
```

Comment:
this test changes config by pointer, so it may affect tests that are running after this